tests/backup/backup_helper.go

```diff
@@ -5996,3 +5995,238 @@ func CreateKubevirtBackupRuleForAllVMsInNamespace(ctx context.Context, namespace
 
 	return ruleStatus, ruleName, nil
 }
+
+// UpdateKDMPConfigMap updates the KDMP configMap with the given key and value.
+func UpdateKDMPConfigMap(dataKey string, dataValue string) error {
+	KDMPconfigMapName := "kdmp-config"
+	KDMPconfigMapNamespace := "kube-system"
+	KDMPconfigMap, err := core.Instance().GetConfigMap(KDMPconfigMapName, KDMPconfigMapNamespace)
+	if err != nil {
+		return err
+	}
+	KDMPconfigMap.Data[dataKey] = dataValue
+	_, err = core.Instance().UpdateConfigMap(KDMPconfigMap)
+	if err != nil {
+		return err
+	}
+	log.Infof("updated the KDMP configMap data with key [%s]: value [%s]", dataKey, dataValue)
+	return nil
+}
+
+type PodDirectoryConfig struct {
+	BasePath          string
+	Depth             int
+	Levels            int
+	FilesPerDirectory int
+	FileSizeInMB      int
+	FileName          string
+	DirName           string
+}
+
+// CreateNestedDirectoriesWithFilesInPod creates a nested directory structure with files within a specified Pod.
+func CreateNestedDirectoriesWithFilesInPod(pod corev1.Pod, directoryConfig PodDirectoryConfig) error {
+	var wg sync.WaitGroup
+	var errChan = make(chan error)
+	fileConfig := PodDirectoryConfig{
+		BasePath:          directoryConfig.BasePath,
+		FilesPerDirectory: directoryConfig.FilesPerDirectory,
+	}
+	err := CreateFilesInPodDirectory(pod, fileConfig)
+	if err != nil {
+		return fmt.Errorf("error creating files in %s: %s", directoryConfig.BasePath, err.Error())
+	}
+
+	if directoryConfig.Levels > 0 && directoryConfig.Depth > 0 {
+		for i := 0; i < directoryConfig.Depth; i++ {
+			dirPath := filepath.Join(directoryConfig.BasePath, fmt.Sprintf("level_%d_depth_%d_%s", directoryConfig.Levels, i, RandomString(4)))
+			podConfig := PodDirectoryConfig{
+				BasePath: dirPath,
+			}
+			err := CreateDirectoryInPod(pod, podConfig)
+			if err != nil {
+				log.Infof(fmt.Sprintf("Error creating directory %s: %s\n", dirPath, err.Error()))
+				continue
+			}
+			fileConfig := PodDirectoryConfig{
+				BasePath:          dirPath,
+				FilesPerDirectory: directoryConfig.FilesPerDirectory,
+			}
+
+			err = CreateFilesInPodDirectory(pod, fileConfig)
+			if err != nil {
+				return fmt.Errorf("error creating files in %s: %s", dirPath, err.Error())
+			}
+
+			if directoryConfig.Levels > 1 {
+				wg.Add(1)
+				go func(dirPath string, depth int) {
+					defer wg.Done()
+					DirectoryConfig := PodDirectoryConfig{
+						BasePath:          dirPath,
+						Depth:             directoryConfig.Depth,
+						Levels:            directoryConfig.Levels - 1,
+						FilesPerDirectory: directoryConfig.FilesPerDirectory,
+					}
+					errChan <- CreateDirectoryStructureInPod(pod, DirectoryConfig)
+				}(dirPath, directoryConfig.Depth)
+			}
+		}
+	}
+	go func() {
+		wg.Wait()
+		close(errChan)
+	}()
+	for err := range errChan {
+		if err != nil {
+			return fmt.Errorf("error creating nested directories: %s", err.Error())
+		}
+	}
```

Comment:
How about? If errChan contains more than one error, we might lose some of the error info.

```go
var errs []error
for err := range errChan {
    if err != nil {
        errs = append(errs, err)
    }
}
if len(errs) > 0 {
    return fmt.Errorf("error creating nested directories: %v", errs)
}
```